specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-05-01/managedClusters.json

```diff
@@ -3317,6 +3320,10 @@
         "privateDNSZone": {
           "type": "string",
           "description": "Private dns zone mode for private cluster. "
+        },
+        "enablePrivateClusterPublicFQDN": {
```

Comment:
ARM recommendation is to change the API version when adding new properties to an existing API. It's not a hard requirement, but it can break down-level clients, depending on the service behavior. For example:
1. Portal or new client PUTs the resource, including the new properties
2. Downlevel client PUTs/PATCHs the same resource without the new properties.

What state will the resource be in afterward? Will it make sense to customers from the perspective of both clients and the portal?
Without incrementing the version, your service can't control that situation, e.g. rejecting down-level PUT/PATCH.

API changes that require a version change: https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#api-changes-that-require-a-version-change

Reply:
Hi @mentat9, this is a new api version 2021-05-01. It's a new property for this new api.

Reply:
OK that's fine then. You checked the box that said you had copied the previous files to the new API version folder before making any modifications. Since the API version change isn't here in the PR, I though you were updating the existing one.